gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/PlanServiceImpl.java

```diff
@@ -692,6 +698,23 @@ private void reorderedAndSavePlansAfterRemove(final Plan planRemoved) throws Tec
             });
     }
 
+    private void validateNoConflictingAuthenticationForNativePlan(final Plan nativePlanToPublish, final Set<Plan> apiPlans) {
+        var planToPublishIsKeyless = nativePlanToPublish.getSecurity() == Plan.PlanSecurityType.KEY_LESS;
+
+        Function<Plan, Boolean> conflictingPublishedPlanSecurity = planToPublishIsKeyless
+            ? plan1 -> plan1.getSecurity() != Plan.PlanSecurityType.KEY_LESS
```

Comment:
Maybe we can find a better name than `plan1`. Why not `existingPlan`?

Reply:
what about 2 filters 🤔 
```
    .filter(existingPlan -> existingPlan.getStatus() == Plan.Status.PUBLISHED)
    .filter(existingPlan -> planToPublishIsKeyless 
        ? existingPlan.getSecurity() != Plan.PlanSecurityType.KEY_LESS 
        : existingPlan.getSecurity() == Plan.PlanSecurityType.KEY_LESS)
```
to avoid Function 
?